agent/pipeline_uploader_test.go

```diff
@@ -0,0 +1,208 @@
+package agent_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/buildkite/agent/v3/agent"
+	"github.com/buildkite/agent/v3/api"
+	"github.com/buildkite/agent/v3/clicommand"
+	"github.com/buildkite/agent/v3/logger"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAsyncPipelineUpload(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	l := clicommand.CreateLogger(&clicommand.PipelineUploadConfig{LogLevel: "notice"})
+
+	type Test struct {
```

Comment:
But it does conflict with the argument to the closure used to avoid loop variable shenanigans: https://go.dev/play/p/VmzVE01PLdl

The type identifier does not need to exist. It's only because of the loop variable shenanigans that we're even having this conversation. I guess just reassigning it avoids all this, so I'll do that.